File: JS/ExpressJSOverview/models/Calculator.js
/*jshint esversion: 6*/
function Calculator(leftOperand, operator, rightOperand) {
    this.leftOperand = leftOperand;
    this.operator = operator;
    this.rightOperand = rightOperand;

    this.calculateResult = function() {
        let result = 0;

        switch (this.operator) {
            case '+':
                result = this.leftOperand + this.rightOperand;
                break;
            case '-':
                result = this.leftOperand - this.rightOperand;
                break;
            case '*':
                result = this.leftOperand * this.rightOperand;
                break;
            case '/':
                result = this.leftOperand / this.rightOperand;
                break;
        }
        return result;
    };
}
// const myCaklk = new Calculator('+', 5, 3);   // UNIT TEST :))
// console.log(myCaklk.calculate());

module.exports = Calculator;

File: CS/CalculatorApp/Models/Calculator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CalculatorApp.Models
{
    public class Calculator
    {
        public decimal LeftOperand { get; set; }
        public decimal RightOperand { get; set; }
        public string Operator { get; set; }
        public decimal Result { get; set; }

        public Calculator()
        {
            this.Result = 0;
        }

        internal void CalculateResult()
        {
            if (this.Operator=="+")
            {
                this.Result = this.LeftOperand + this.RightOperand;
            }
            else if (this.Operator == "-")
            {
                this.Result = this.LeftOperand - this.RightOperand;
            }
            else if (this.Operator == "*")
            {
                this.Result = this.LeftOperand * this.RightOperand;
            }
            else if (this.Operator == "/")
            {
                this.Result = this.LeftOperand / this.RightOperand;
            }
        }
    }
}
